### ISSUE TITLE
object-shorthand rule's consistent-as-needed option does not work for certain inputs

### ISSUE THREAD

**Issue by ahuth** [eslint/eslint#7214]
It appears that the object-shorthand rule has a bug. Specifically, the consistent-as-needed option is not functioning properly.

The option is supposed to enforce consistent use of object shorthand syntax per object literal, and to require shorthand syntax when otherwise we would have redundant key/values.

For instance, instead of

```js
const test = {
  a: a,
  b: b
};
```

we should be required to use

```js
const test = {
  a,
  b
};
```

The following is still allowed

```js
const test = {
  first: 1,
  second: 2
};
```

However, when using a quoted string for the key, OR a number for the key, such as:

```js
const test = {
  0: "zero",
  'quoted-string': "quotes"
};
```

The rule complains that we should be using shorthand. Obviously, that's not possible.

@naomiajacobs and I will submit a PR to resolve this issue.

* **ESLint Version:**: 3.3.1
* **Node Version:**: 6.4.0
* **npm Version:**: 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**

Default

**Please show your full configuration:**

```js
module.exports = {
  extends: 'airbnb',
  rules: {
    'import/no-extraneous-dependencies': [
      'off',
      { devDependencies: true, optionalDependencies: false, peerDependencies: false },
    ],
    'import/no-unresolved': 'off',
    'func-names': 'error',
    'object-shorthand': ['error', 'consistent-as-needed'],
  },
  env: {
    browser: true,
    node: true,
    jasmine: true,
  },
  globals: {
    _: true,
    $: true,
  },
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
const example = {
  '-10': 'low',
  0: 'normal',
  10: 'high',
  20: 'critical',
};
```

**What did you expect to happen?**

This example is not using redudant property/key values, and should pass the consistent-as-needed option for the object-shorthand rule.

**What actually happened? Please include the actual, raw output from ESLint.**

```
error  Expected shorthand for all properties  object-shorthand
```

**Comment by ahuth** [comment]
Fixed in #7215

**Comment by mysticatea** [comment]
Thank you for the report and the fix!

I confirmed it.

**Comment by kaicataldo** [comment]
Looks like this was actually fixed by https://github.com/eslint/eslint/pull/7215. The reason it didn't close was because the commit message is missing `fixes`.
